scripts/modules/events/backward-compatibility.js

```
@@ -0,0 +1,430 @@
+/**
+ * Extract event-specific data from legacy payload
+ *
+ * @param {string} eventType - Event type
+ * @param {Object} eventData - Event data
+ * @returns {Object} Extracted data
+ */
+function extractEventSpecificData(eventType, eventData) {
+	const extracted = { ...eventData };
+
+	// Remove standard fields that are now top-level
+	delete extracted.timestamp;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)